src/MongoDB.Driver.Core/Core/Authentication/ScramShaAuthenticator.cs

```diff
@@ -102,7 +102,7 @@ public abstract class ScramShaAuthenticator : SaslAuthenticator
         public override string DatabaseName => _databaseName;
 
         // nested classes
-        private class ScramShaMechanism : ISaslMechanism
+        internal class ScramShaMechanism : ISaslMechanism
```

Comment:
Need to promote this to internal so `SaslAuthenticator` can access this nested class

Reply:
Would not be necessary if you adopt my suggestion above.